tests/modcc/test_kinetic_rewriter.cpp

```diff
@@ -0,0 +1,507 @@
+#include "test.hpp"
+
+#include <cmath>
+#include <stdexcept>
+#include <string>
+
+#include "expression.hpp"
+#include "kinrewriter.hpp"
+#include "parser.hpp"
+
+// Simple algebraic term expansion/collection routines.
+
+namespace alg {
```

Comment:
I've historically indented 'internal' namespaces, but wouldn't indent the namespace that everything is in. This is a totally arbitrary distinction, and I'm not going to defend it.

Putting it into its own header is a good idea. I will move it out.